src/openslide-vendor-dicom.c

```diff
@@ -718,9 +689,33 @@ static bool maybe_add_file(openslide_t *osr,
     return true;
   }
 
-  // check the image format
-  if (!get_format(f->metadata, &f->format, err)) {
-    return false;
+  // check the other image format tags, ignore DICOMs with formats we don't
+  // support
+  switch (f->format) {
+  case FORMAT_JPEG2000:
+    if (verify_tag_str(f->metadata, PhotometricInterpretation, "YBR_ICT")) {
+      f->jp2k_colourspace = OPENSLIDE_JP2K_YCBCR;
+    } else if (verify_tag_str(f->metadata, PhotometricInterpretation, "RGB")) {
+      f->jp2k_colourspace = OPENSLIDE_JP2K_RGB;
+    } else {
+      return true;
+    }
+    break;
+  case FORMAT_JPEG:
+    if (!verify_tag_str(f->metadata, PhotometricInterpretation, "YBR_FULL_422") &&
+        !verify_tag_str(f->metadata, PhotometricInterpretation, "RGB")) {
+      return true;
+    }
+  default:
+    break;
+  }
+  if (!verify_tag_int(f->metadata, PlanarConfiguration, 0) ||
+      !verify_tag_int(f->metadata, BitsAllocated, 8) ||
+      !verify_tag_int(f->metadata, BitsStored, 8) ||
+      !verify_tag_int(f->metadata, HighBit, 7) ||
+      !verify_tag_int(f->metadata, SamplesPerPixel, 3) ||
+      !verify_tag_int(f->metadata, PixelRepresentation, 0)) {
```

Comment:
Nit: I'd be inclined to put the generic checks first.

Reply:
OK, done.